packages/tools/webpack/extract-depndencies-webpack-plugin.js

```diff
@@ -2,54 +2,117 @@
-		this.replaceDependencyNames = replaceDependencyNames || this.defaultReplaceDependencyNames;
-		this.generateAssetsFileName = generateAssetsFileName || this.defaultGenerateAssetsFileName;
+		if ( ! handlePrefix ) {
+			throw new Error( 'handlePrefix is required in Webpack.config' );
+		}
+
+		this.handlePrefix = handlePrefix;
+		this.i18n = i18n;
 	}
 
 	apply( compiler ) {
 		compiler.hooks.thisCompilation.tap( this.constructor.name, ( compilation ) => {
 			compilation.hooks.processAssets.tap( { name: this.constructor.name }, () => {
-				[ ...compilation.entrypoints ].forEach( ( [ id, entrypoint ] ) => {
-					const chunk = entrypoint.chunks.find( ( { name } ) => name === id );
+				const handlesAssetsMap = [ ...compilation.entrypoints ].reduce( ( map, [ entryName, entrypoint ] ) => {
+					const chunk = entrypoint.chunks.find( ( { name } ) => name === entryName );
 					const chunkJSFile = this.getFileFromChunk( chunk );
 
 					if ( ! chunkJSFile ) {
-						return;
+						return map;
 					}
 
+					const rootFolderPath = compilation.options.context;
+					const entryFolderPath = entrypoint.origins[ 0 ].request;
+					const packageJsonPath = this.findPackageJsonPath( entryFolderPath, rootFolderPath );
+
+					if ( ! packageJsonPath ) {
+						return map;
+					}
+
+					const packageJsonData = require( packageJsonPath );
+
 					const deps = this.getDepsFromChunk( compilation, chunk );
 
 					const assetFilename = this.generateAssetsFileName(
 						compilation.getPath( '[file]', { filename: chunkJSFile } )
 					);
 
-					const content = this.createAssetsFileContent(
-						id,
+					const handle = this.generateHandleName( entryName );
+
+					map[ handle ] = assetFilename;
+
+					const content = this.createAssetsFileContent( {
 						deps,
-						compilation.options.optimization.minimize
-					);
+						entryName,
+						i18n: this.i18n,
+						type: packageJsonData.elementor?.type,
+					} );
 
 					// Add source and file into compilation for webpack to output.
 					compilation.assets[ assetFilename ] = new RawSource( content );
 
 					chunk.files.add( assetFilename );
-				} );
+
+					return map;
+				}, {} );
+
+				const loaderFileContent = this.getLoaderFileContent( handlesAssetsMap );
+
+				compilation.emitAsset(
+					'loader.php',
+					new RawSource( loaderFileContent )
+				);
 			} );
 		} );
 	}
 
+	findPackageJsonPath( baseFolder, topLevelFolder ) {
+		const packageJsonPath = path.resolve( baseFolder, 'package.json' );
+
+		if ( fs.existsSync( packageJsonPath ) ) {
+			return packageJsonPath;
+		}
+
+		const parentFolder = path.resolve( baseFolder, '..' );
+
+		if ( parentFolder === topLevelFolder ) {
+			return null;
+		}
+
+		return this.findPackageJsonPath( parentFolder, topLevelFolder );
+	}
+
+	getLoaderFileContent( entriesData ) {
+		const entriesContent = Object.entries( entriesData ).map( ( [ handle, assetFileName ] ) => {
+			return `
+	$data['${ handle }'] = require __DIR__ . '/${ assetFileName }';`;
```

Comment:
How you can be sure all the output files are in the same folder + I second though it feels little bit scary and can potential lead to some bugs

Reply:
As agreed, it's ok (for now) .